storm-server/src/main/java/org/apache/storm/metric/StormMetricsRegistry.java

```diff
@@ -12,79 +12,122 @@
+    private static final StormMetricsRegistry DEFAULT_REGISTRY = new StormMetricsRegistry();
     private static final Logger LOG = LoggerFactory.getLogger(StormMetricsRegistry.class);
 
-    public static Meter registerMeter(String name) {
-        Meter meter = new Meter();
-        return register(name, meter);
-    }
+    private StormMetricsRegistry() {/*Singleton pattern*/}
 
-    // TODO: should replace Callable to Gauge<Integer> when nimbus.clj is translated to java
-    public static Gauge<Integer> registerGauge(final String name, final Callable fn) {
-        Gauge<Integer> gauge = new Gauge<Integer>() {
-            @Override
-            public Integer getValue() {
-                try {
-                    return (Integer) fn.call();
-                } catch (Exception e) {
-                    LOG.error("Error getting gauge value for {}", name, e);
-                }
-                return 0;
+    /**
+     * Register a gauge with provided callback. This swallows all exceptions
+     * thrown from the callback, consider using {@link #registerProvidedGauge(String, Gauge)}
+     * if no exceptions will be thrown by the callable.
+     *
+     * @param name name of the gauge
+     * @param fn callback that measures
+     * @param <V> type of measurement the callback returns, also the type of gauge
+     * @return registered gauge
+     */
+    public static <V> Gauge<V> registerGauge(final String name, final Callable<V> fn) {
+        return DEFAULT_REGISTRY.register(name, () -> {
+            try {
+                return fn.call();
+            } catch (Exception e) {
+                LOG.error("Error getting gauge value for {}", name, e);
             }
-        };
-        return register(name, gauge);
+            return null;
+        });
+    }
+
+    /**
+     * Register a provided gauge. Use this method if a custom gauge is needed or
+     * exceptions have already been handled.
+     *
+     * @param name name of the gauge
+     * @param gauge gauge
+     * @param <V> type of value the gauge measures
+     */
+    public static <V> void registerProvidedGauge(final String name, final Gauge<V> gauge) {
+        DEFAULT_REGISTRY.register(name, gauge);
     }
 
-    public static void registerProvidedGauge(final String name, Gauge gauge) {
-        register(name, gauge);
+    public static Histogram registerHistogram(String name) {
+        return registerHistogram(name, new ExponentiallyDecayingReservoir());
     }
 
     public static Histogram registerHistogram(String name, Reservoir reservoir) {
-        Histogram histogram = new Histogram(reservoir);
-        return register(name, histogram);
+        return DEFAULT_REGISTRY.register(name, new Histogram(reservoir));
     }
 
+    public static Meter registerMeter(String name) {
+        return DEFAULT_REGISTRY.register(name, new Meter());
+    }
+
+    //Change the name to avoid name conflict in future Metrics release
```

Comment:
There couldn't be a name conflict, because this method is static, and the one on the super class isn't.

Reply:
You are right, my bad.